lib/streamlit/elements/multiselect.py

```diff
@@ -216,14 +216,17 @@ def _check_and_convert_to_indices(
                 else:
                     default_values = list(default_values)
             if len(default_values) != 0 and isinstance(default_values[0], Enum):
```

Comment:
On second thought, I suggest dropping this conditional entirely, and rather check for enums inside the check you're going to do below regardless. That check would then become:

```py
for value in default_values:
    if value not in opt:
        if isinstance(value, Enum):
            raise StreamlitAPIException(
                """Every Multiselect default value must exist in options. 
                If this looks incorrect and you're not using enums in a 
                separate module, we recommend you do in order to remove 
                this log statement and fix enums mysteriously not being equal. 
                This occurs because of Streamlit's built-in mechanic of rerun 
                thus causing the memory addresses of the variants of the enum
                to differ from script run to script run."""
            )
        raise StreamlitAPIException(
            "Every Multiselect default value must exist in options"
        )
```

Reply:
If you also would like to avoid raising the longer exception string when the user is using `Enum` values, but they've actually made a mistake, and the default value shouldn't exist in the options regardless of reruns, you can check with:

```py
def likely_equivalent(e1: Enum, e2: object) -> bool:
    """Checks whether two enum values are likely to have been identical if it wasn't for rerunning.
    """
    return (
        isinstance(e2, Enum) and
        type(e2).__name__ == type(e1).__name__ and
        e2.name == e1.name and
        e2.value == e1.value
    )

...

if isinstance(value, Enum) and all(not likely_equivalent(value, option) for option in opt):
    ...
```

Reply:
Why is the second object for likely_equivalent an object? can't we just make it an enum?

Reply:
That would assume all objects in `opt` are of Enum type. That would be the most common case, but it isn't guaranteed. This function should be able to compare the Enum value to any object.

Reply:
Not sure what I was thinking. You of course want to raise with the long message if the following check holds:

```py
if isinstance(value, Enum) and any(likely_equivalent(value, option) for option in opt):
```

Or, in plain English: If there are values in opt that look like they might be duplicates of value, due to re-running.

Sorry for the confusion. Was thinking too fast.